### DISCUSSION TITLE
Webscraping?

### DISCUSSION THREAD

**Discussion by EpicLPer** [mampfes/hacs_waste_collection_schedule#1137] [Q&A] [answered]
Heya,

my local waste calendar is not available as any kind of file, but they're updating dates on a website or PDF.
Is there any way to scrape that info automatically and add it somehow?

Thanks!

**Comment by dt215git** [comment]
If you provide a link to the website, and an example address, someone can take a look and see what's possible.

**Comment by EpicLPer** [comment]
Heya, the info is available here: https://scheibbs.umweltverbaende.at/?kat=32

They only provide PDFs or website data, I've already contacted them and asked if they could provide one in an ical format, tho except for a mail asking me what "ical even is" I didn't get another reply yet :)

**Reply by dt215git** [reply, marked as answer]
No need to parse the pdf, the dates on the web pages can be scrapped.
PR #1157 created.

**Reply by EpicLPer** [reply]
Oh, awesome! Thanks a "ton" heh 👌

**Reply by 5ila5** [reply]
added by #1157 

will be part of release 1.42.0 or is available now as master/git version (up to date clone of this repository)

**Comment by AghilasMessara** [comment]
Hello,
If I understand the PR, this has been added for a specific city. Is there a way for us to have a generic web scrapper that we can configure or even like code a local plugin in our homeassistant?
I feel bad asking to add code that most likely only be used by me 😆

**Reply by 5ila5** [reply]
No there is currently no web scraping source. If you can get an ICS/ICAL file with web requests you can use the [ICS source](https://github.com/mampfes/hacs_waste_collection_schedule/blob/master/doc/source/ics.md). Feel free to ask for your city to be supported (open Issue) or create a source yourself [contribution guide](https://github.com/mampfes/hacs_waste_collection_schedule/blob/master/doc/contributing.md) ([for dedicated sources, without ICS support](https://github.com/mampfes/hacs_waste_collection_schedule/blob/master/doc/contributing_source.md)) normally it's not that complicated if you have some python experience. Don't feel bad for requesting your city, there are a lot of sources to support smaller areas or cities and they are typically not that much work. Goal of this project is to support as many areas as possible. Maybe someone else also want to use the source as well.

**Reply by AghilasMessara** [reply]
Ok thank you I will try a PR then. 👍

**Reply by 5ila5** [reply]
btw. It's best to first check weather a similar source is already supported. for me the easiest way is to search for files/addresses the website makes requests (e.g. `.../apibroker/runLookup`). 

If you do need help feel free to ask or open a draft pull request.

**Reply by AghilasMessara** [reply]
Will do. The README doesn't state any source from France so I guess my city is not already there.
